File: Assigment 3/C/04. Sparse matrix or not .c
#include <stdio.h>

int main(){
    int arr[4][5] 
    = {
        {0 , 0 , 3 , 0 , 4 },
        {0 , 0 , 5 , 7 , 0 },
        {0 , 0 , 0 , 0 , 0 },
        {0 , 2 , 6 , 0 , 0 }
    };
    int r = sizeof(arr)/sizeof(arr[0]);
    int c = sizeof(arr[0])/sizeof(arr[0][0]);

    int count = 0;

    for(int i=0; i<r; i++){
        for(int j=0; j<c; j++){
            if(arr[i][j]==0){
                count++;
            }
        }
    }

    int t = (r*c)/2;

    if(count>t){
        printf("sparse matrix");
    }else{
        printf("not");
    }

    return 0;
}

File: Assigment 3/Python/04. Sparse matrix or not .py

arr = [
    [0, 0, 3, 0, 4],
    [0, 0, 5, 7, 0],
    [0, 0, 0, 0, 0],
    [0, 2, 6, 0, 0]
]

r = len(arr)
c = len(arr[0])

z = 0
for i in range(r):
    for j in range(c):
        if(arr[i][j]==0):
            z+=1

if(z>((r*c)/2)):
    print("sparse matrix")
else:
    print("not")
    